gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java

```diff
@@ -109,28 +109,27 @@ public Blob(Storage storage, BlobInfo info) {
   }
 
   /**
-   * Constructs a {@code Blob} object for the provided bucket and blob names. The storage service is
-   * used to issue requests.
+   * Creates a {@code Blob} object for the provided bucket and blob names. Performs an RPC call to
+   * get the latest blob information.
    * 
    * @param storage the storage service used for issuing requests
    * @param bucket bucket's name
    * @param blob blob's name
    */
-  public Blob(Storage storage, String bucket, String blob) {
-    this.storage = checkNotNull(storage);
-    this.info = BlobInfo.builder(BlobId.of(bucket, blob)).build();
+  public static Blob load(Storage storage, String bucket, String blob) {
+    return load(storage, BlobId.of(bucket, blob));
   }
 
   /**
-   * Constructs a {@code Blob} object for the provided {@code BlobId}. The storage service is used
-   * to issue requests.
+   * Creates a {@code Blob} object for the provided {@code blobId}. Performs an RPC call to get the
+   * latest blob information.
    * 
    * @param storage the storage service used for issuing requests
    * @param blobId blob's identifier
    */
-  public Blob(Storage storage, BlobId blobId) {
-    this.storage = checkNotNull(storage);
-    this.info = BlobInfo.builder(blobId).build();
+  public static Blob load(Storage storage, BlobId blobId) {
+    BlobInfo info = storage.get(blobId);
```

Comment:
Same comment for here and bucket:
(1)  javadoc should include "@throws StorageException upon failure"
(2) NPE if get returns `null`